extensions/wikia/Search/templates/WikiaSearch_fandomStories.php

```diff
@@ -0,0 +1,31 @@
+<?php $counter = 0; ?>
+
+<?php if ( !empty( $stories ) ) : ?>
+	<div class="side-articles fandom-stories RailModule">
+		<h1 class="side-articles-header"><?= wfMessage( 'wikiasearch2-fandom-stories-title', count( $stories ) ); ?></h1>
+		<div>
+		<?php foreach ( $stories as $story ) : ?>
+			<div class="side-article side-article-<?= mb_strtolower($story['vertical'], 'UTF-8') ?> result">
+				<div class="side-article-category"><?= $story['vertical'] ?></div>
+				<div class="side-article-thumbnail">
+					<? if ( isset( $story['image'] ) ) : ?>
+						<a href="<?=$story['url']?>" class="fandom-story-image" data-pos="<?= $counter ?>">
+							<img src="<?= $story['image'] ?>" />
+						</a>
+					<? endif; ?>
+				</div>
+				<div class="side-article-text">
+					<a href="<?= $story['url'] ?>" class="fandom-story-link" data-pos="<?= $counter ?>"><?= $story['title'] ?></a>
+					<div class="fandom-annotation">fandom</div>
+				</div>
+			</div>
+		<?php $counter++; ?>
+		<?php endforeach; ?>
+		</div>
+		<?php if ( !empty( $viewMoreLink ) ) : ?>
+			<div class="side-articles-footer">
+				<a href="<?= $viewMoreLink ?>"><?= wfMessage( 'wikiasearch2-fandom-stories-view-more' ); ?>  <?= DesignSystemHelper::getSvg( 'wds-icons-arrow', 'wds-icon wds-icon-small fandom-stories__arrow' ); ?></a>
```

Comment:
`viewMoreLink` is vulnerable to XSS via direct wikia.php calls to the `fandomStories` controller method.